miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/MetadataApi.kt

```diff
@@ -0,0 +1,37 @@
+package com.rakuten.tech.mobile.miniapp.api
+
+import androidx.annotation.Keep
+import com.google.gson.annotations.SerializedName
+import retrofit2.Call
+import retrofit2.http.GET
+import retrofit2.http.Path
+
+internal interface MetadataApi {
+    @GET("host/{hostappId}/miniapp/{miniappId}/version/{versionId}/{testPath}/metadata")
+    fun fetchMetadata(
+        @Path("hostappId") hostAppId: String,
+        @Path("miniappId") miniAppId: String,
+        @Path("versionId") versionId: String,
+        @Path("testPath") testPath: String = ""
+    ): Call<MetadataEntity>
+}
+
+@Keep
+internal data class MetadataEntity(
+    @SerializedName("metadata") val metadata: MetadataResponse
```

Comment:
I'm not sure if we can `MetadataResponse` directly. The Host App wants to have access to all key/values of the manifest through `MiniAppManifest.manifest`. But that does make things more complicated because we would need to parse this as `Map<String, String>`...

Maybe we should re-think our approach to custom keys for the Host App. Instead of giving access to the whole manifest through `MiniAppManifest.manifest`, could we add a key like `metadata` where the Host App can use any custom keys?

Reply:
Yeah, currently I have something in this part with `customMetaData`, but I doubt that is not enough useful for hostapp.

Reply:
I have updated this part.